app/components/sidebars/main/channels_list/list/index.js

```diff
@@ -49,7 +51,16 @@ function mapStateToProps(state) {
         sidebarPrefs.favorite_at_top === 'true' && favoriteChannelIds.length,
     ));
 
+    let canJoinPublicChannels = true;
+    if (hasNewPermissions(state)) {
```

Comment:
Not specific to this PR, but I wonder when we're going to release the new ESR. It would be really nice to be able to remove these `hasNewPermissions` checks